### PR TITLE
feat: added flag

### PR DESCRIPTION
**Pull request by 0xZenus**
# Title

Please include a title.

## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

### CODE REVIEW

File: packages/api/src/v1-queries/get-portfolio-positions/getV1PortfolioPositionDetails.ts [hunk 1]
```diff
@@ -70,7 +71,7 @@ export const getV1PortfolioPositionDetails = async ({
       realizedPNLCashflow,
       realizedPNLTotal: realizedPNLCashflow + response.realizedPNLFees,
 
-      history: txs,
+      history: includeHistory || response.pool.flags.isGLP28Jun2023 ? txs : [],
```
Comment (ssbarbee) [inline packages/api/src/v1-queries/get-portfolio-positions/getV1PortfolioPositionDetails.ts:74]: Don't do the logic here. Why even bother to push items to txs if we dont need them? Save some CPU time man :) Here it should always be history: txs but on upper level you determine if array needs to be built!

File: packages/api/src/v1-queries/get-portfolio-positions/getV1PortfolioPositionDetails.ts [hunk 2]
```diff
@@ -107,7 +108,7 @@ export const getV1PortfolioPositionDetails = async ({
       canSettle: true,
       rolloverPoolId,
 
-      history: txs,
+      history: includeHistory || response.pool.flags.isGLP28Jun2023 ? txs : [],
```
Comment (ssbarbee) [inline packages/api/src/v1-queries/get-portfolio-positions/getV1PortfolioPositionDetails.ts:111]: Don't do the logic here. Why even bother to push items to txs if we dont need them? Save some CPU time man :) Here it should always be history: txs but on upper level you determine if array needs to be built!

File: packages/api/src/v1-queries/get-portfolio-positions/getV1PortfolioPositionDetails.ts (hunk 1 above)
Comment (0xZenus) [reply packages/api/src/v1-queries/get-portfolio-positions/getV1PortfolioPositionDetails.ts]: https://github.com/Voltz-Protocol/v2-off-chain-monorepo/pull/126/commits/972f309bd5b012bea2cc64b798f4c754a2fdd8e4

File: packages/api/src/v1-queries/get-portfolio-positions/getV1PortfolioPositionDetails.ts (hunk 2 above)
Comment (0xZenus) [reply packages/api/src/v1-queries/get-portfolio-positions/getV1PortfolioPositionDetails.ts]: https://github.com/Voltz-Protocol/v2-off-chain-monorepo/pull/126/commits/972f309bd5b012bea2cc64b798f4c754a2fdd8e4

File: packages/api/src/v1-queries/get-portfolio-positions/getV1PortfolioPositionDetails.ts [hunk 3]
```diff
@@ -118,6 +119,6 @@ export const getV1PortfolioPositionDetails = async ({
     canSettle: false,
     rolloverPoolId: null,
 
-    history: txs,
+    history: includeHistory || response.pool.flags.isGLP28Jun2023 ? txs : [],
```
[the remark this answers is not in the document]
Comment (0xZenus) [reply packages/api/src/v1-queries/get-portfolio-positions/getV1PortfolioPositionDetails.ts]: https://github.com/Voltz-Protocol/v2-off-chain-monorepo/pull/126/commits/972f309bd5b012bea2cc64b798f4c754a2fdd8e4

**Review by ssbarbee — APPROVED** [review]

**Pull request merged**